### PR TITLE
add method to persist tag files to deposit record in redis

### PR DESCRIPTION
**Pull request by harringj**
We added a new method that allows us to keep a bag's tag files in the deposit record.

### CODE REVIEW

File: metadata/src/main/java/edu/unc/lib/dl/util/DepositStatusFactory.java [hunk 1]
```diff
@@ -4,9 +4,12 @@
 import static edu.unc.lib.dl.util.RedisWorkerConstants.DEPOSIT_STATUS_PREFIX;
 import static edu.unc.lib.dl.util.RedisWorkerConstants.INGESTS_CONFIRMED_PREFIX;
 import static edu.unc.lib.dl.util.RedisWorkerConstants.INGESTS_UPLOADED_PREFIX;
+import static edu.unc.lib.dl.util.RedisWorkerConstants.DEPOSIT_MANIFEST_PREFIX;
```
Comment (bbpennel) [inline metadata/src/main/java/edu/unc/lib/dl/util/DepositStatusFactory.java]: I'm not seeing this constant being declared in this pull request

Comment (harringj) [inline metadata/src/main/java/edu/unc/lib/dl/util/DepositStatusFactory.java]: Good catch. Not sure why git didn't pick up that file, but I re-committed and push out the change,

**Comment by bbpennel** [comment]
For this to function i believe you will need to change the MakeFOXML side of things as well, and also change the METS normalization job so everything works the same way

File: deposit/src/main/java/edu/unc/lib/deposit/fcrepo3/MakeFOXML.java [hunk 2]
```diff
@@ -343,14 +344,19 @@ private void writeDepositRecord(Resource deposit) {
...
-					"M", manifest.toURI().toString(), "text/xml", "URL", dsLabel, false, null);
-			foxml.getRootElement().addContent(el);
-			log.info("Manifest file exists and has been added");
+		String dsLabel = Datastream.DATA_MANIFEST.getLabel();
+		ArrayList<File> manifestFiles = getManifestFiles();
+		if (!manifestFiles.isEmpty()) {
+			int i = 0;
+			Element el ;
+			for (File manifest : manifestFiles) {
+			el	= FOXMLJDOMUtil.makeLocatorDatastream(Datastream.DATA_MANIFEST.getName() + i++,
```
Comment (bbpennel) [inline deposit/src/main/java/edu/unc/lib/deposit/fcrepo3/MakeFOXML.java]: Tab ended up on the wrong side of "el" here

Comment (harringj) [inline deposit/src/main/java/edu/unc/lib/deposit/fcrepo3/MakeFOXML.java]: Fixed!

File: deposit/src/main/java/edu/unc/lib/deposit/fcrepo3/MakeFOXML.java [hunk 3]
```diff
@@ -343,14 +344,19 @@ private void writeDepositRecord(Resource deposit) {
...
+						"M", manifest.getAbsolutePath(), "text/xml", "URL", dsLabel, false, null);
+				foxml.getRootElement().addContent(el);
+				log.info("Manifest files have been added to foxml");
+			} 
+		} else {
+			log.warn("No manifest files were found for the deposit");
```
Comment (bbpennel) [inline deposit/src/main/java/edu/unc/lib/deposit/fcrepo3/MakeFOXML.java]: We will likely get this warning for all deposits besides bags and METS.  I think we may want to remove it until we come to a point where all deposit types should be adding manifest files

File: deposit/src/main/java/edu/unc/lib/deposit/work/AbstractDepositJob.java [hunk 4]
```diff
@@ -1,16 +1,17 @@
 package edu.unc.lib.deposit.work;
 
 import static edu.unc.lib.dl.util.DepositConstants.DESCRIPTION_DIR;
-import static edu.unc.lib.dl.util.RedisWorkerConstants.DepositField.manifestURI;
-
+import static edu.unc.lib.dl.util.RedisWorkerConstants.DEPOSIT_MANIFEST_PREFIX;
```
Comment (bbpennel) [inline deposit/src/main/java/edu/unc/lib/deposit/work/AbstractDepositJob.java]: This import isn't needed here

File: deposit/src/main/java/edu/unc/lib/deposit/fcrepo3/MakeFOXML.java (hunk 3 above)
Comment (harringj) [inline deposit/src/main/java/edu/unc/lib/deposit/fcrepo3/MakeFOXML.java]: I changed the logging statement to a log.info with wording that suggests the missing manifest is due to the deposit not being a bag or METS submission.

File: deposit/src/main/java/edu/unc/lib/deposit/fcrepo3/MakeFOXML.java [hunk 5]
```diff
@@ -343,14 +343,19 @@ private void writeDepositRecord(Resource deposit) {
...
+		String dsLabel = Datastream.DATA_MANIFEST.getLabel();
+		List<String> filePaths = getManifestFileURIs();
+		if (!filePaths.isEmpty()) {
+			int i = 0;
+			Element el ;
+			for (String path : filePaths) {
+				el	= FOXMLJDOMUtil.makeLocatorDatastream(Datastream.DATA_MANIFEST.getName() + i++,
+						"M", path, "text/xml", "URL", dsLabel, false, null);
+				foxml.getRootElement().addContent(el);
+				log.info("Manifest files have been added to foxml");
```
Comment (bbpennel) [inline deposit/src/main/java/edu/unc/lib/deposit/fcrepo3/MakeFOXML.java]: This log statement should either be moved out of the for loop with the count of the number of manifest files added, or it should read "Manifest file {} has been added to foxml", with the name of the manifest file included there.  The syntax for that is 
log.info("message {}", insertValue)

File: deposit/src/main/java/edu/unc/lib/deposit/fcrepo3/MakeFOXML.java [hunk 6]
```diff
@@ -343,14 +343,19 @@ private void writeDepositRecord(Resource deposit) {
...
-			String dsLabel = Datastream.DATA_MANIFEST.getLabel();
-			Element el = FOXMLJDOMUtil.makeLocatorDatastream(Datastream.DATA_MANIFEST.getName(),
-					"M", manifest.toURI().toString(), "text/xml", "URL", dsLabel, false, null);
-			foxml.getRootElement().addContent(el);
-			log.info("Manifest file exists and has been added");
+		String dsLabel = Datastream.DATA_MANIFEST.getLabel();
+		List<String> filePaths = getManifestFileURIs();
+		if (!filePaths.isEmpty()) {
+			int i = 0;
+			Element el ;
```
Comment (bbpennel) [inline deposit/src/main/java/edu/unc/lib/deposit/fcrepo3/MakeFOXML.java]: remove extra space before ;

File: deposit/src/test/java/edu/unc/lib/deposit/normalize/BagIt2N3BagJobTest.java [hunk 7]
```diff
@@ -84,13 +89,14 @@ public void setup() throws Exception {
...
 		setField(job, "depositsDirectory", depositsDirectory);
 		setField(job, "depositStatusFactory", depositStatusFactory);
-
 		job.init();
 	}
 
 	@Test
 	public void testConversion() throws Exception {
 		status.put(DepositField.sourcePath.name(), "src/test/resources/paths/valid-bag");
+		//BagFactory bagFactory = new BagFactory();
```
Comment (bbpennel) [inline deposit/src/test/java/edu/unc/lib/deposit/normalize/BagIt2N3BagJobTest.java]: Remove new commented out code

File: deposit/src/test/java/edu/unc/lib/deposit/normalize/BagIt2N3BagJobTest.java [hunk 8]
```diff
@@ -136,6 +142,12 @@ public URI answer(InvocationOnMock invocation) throws URISyntaxException {
 			children.put(file.getProperty(dprop(model, label)).getString(), file);
 		}
 		
+		ArgumentCaptor<String> arg = ArgumentCaptor.forClass(String.class);
+		verify(depositStatusFactory, times(2)).addManifest(anyString(), arg.capture());
+		List<String> capturedFilenames = Arrays.asList("bagit.txt", "manifest-md5.txt");
```
Comment (bbpennel) [inline deposit/src/test/java/edu/unc/lib/deposit/normalize/BagIt2N3BagJobTest.java]: This isn't actually testing the captured filenames, its testing against a hand built list of filenames.  capturedFilenames should be asking the ArgumentCaptor for all of the captured values.  You can also compare two arrayLists to each other with .equals

File: deposit/src/main/java/edu/unc/lib/deposit/work/AbstractDepositJob.java [hunk 9]
```diff
@@ -190,15 +188,13 @@ public File getEventsDirectory() {
 	}
 
 	/**
-	 * Returns the file where the manifest for this deposit is stored. If no manifest was set, then null is returned
+	 * Returns the files where the manifest for this deposit is stored. If no manifest was set, then null is returned
```
Comment (bbpennel) [inline deposit/src/main/java/edu/unc/lib/deposit/work/AbstractDepositJob.java]: Update comment to reflect that it is returning manifest uris

File: deposit/src/main/java/edu/unc/lib/deposit/fcrepo3/MakeFOXML.java (hunk 5 above)
Comment (harringj) [inline deposit/src/main/java/edu/unc/lib/deposit/fcrepo3/MakeFOXML.java]: Done!

File: deposit/src/test/java/edu/unc/lib/deposit/normalize/BagIt2N3BagJobTest.java (hunk 7 above)
Comment (harringj) [inline deposit/src/test/java/edu/unc/lib/deposit/normalize/BagIt2N3BagJobTest.java]: Deleted!

File: deposit/src/main/java/edu/unc/lib/deposit/work/AbstractDepositJob.java (hunk 9 above)
Comment (harringj) [inline deposit/src/main/java/edu/unc/lib/deposit/work/AbstractDepositJob.java]: Amended!

File: deposit/src/test/java/edu/unc/lib/deposit/normalize/BagIt2N3BagJobTest.java (hunk 8 above)
Comment (harringj) [inline deposit/src/test/java/edu/unc/lib/deposit/normalize/BagIt2N3BagJobTest.java]: Whoops, my bad! This was left in an intermediate state. It should be fixed now.

File: deposit/src/test/java/edu/unc/lib/deposit/normalize/BagIt2N3BagJobTest.java [hunk 10]
```diff
@@ -31,17 +31,23 @@
...
 import java.net.URI;
 import java.net.URISyntaxException;
+import java.util.Arrays;
 import java.util.HashMap;
 import java.util.HashSet;
+import java.util.List;
 import java.util.Map;
 import java.util.Set;
 
+import org.apache.xerces.dom3.as.ASObjectList;
```
Comment (bbpennel) [inline deposit/src/test/java/edu/unc/lib/deposit/normalize/BagIt2N3BagJobTest.java]: Remove unused import

File: metadata/src/test/java/edu/unc/lib/dl/util/DepositStatusFactoryTest.java [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+import org.junit.Before;
+import org.junit.Test;
+
+import redis.clients.jedis.JedisPool;
+
+public class DepositStatusFactoryTest {
+	
+	private DepositStatusFactory factory;
+	private JedisPool jedisPool;
+	private final String uuid = Integer.toString(new Random().nextInt(99999));
```
Comment (bbpennel) [inline metadata/src/test/java/edu/unc/lib/dl/util/DepositStatusFactoryTest.java]: I know there is only one method currently, uuid and filenames should be local variables in the test method unless they are expected to be shared across test methods.  We would most likely want a different uuid for each test method anyways to make sure they don't interfere

File: metadata/src/test/java/edu/unc/lib/dl/util/DepositStatusFactoryTest.java [hunk 12]
```diff
@@ -0,0 +1,38 @@
+package edu.unc.lib.dl.util;
```
Comment (bbpennel) [inline metadata/src/test/java/edu/unc/lib/dl/util/DepositStatusFactoryTest.java]: Could you add the license header to this file?  You can find it in most of the other cdr classes

File: deposit/src/test/java/edu/unc/lib/deposit/normalize/BagIt2N3BagJobTest.java (hunk 10 above)
Comment (harringj) [inline deposit/src/test/java/edu/unc/lib/deposit/normalize/BagIt2N3BagJobTest.java]: Removed!

File: metadata/src/test/java/edu/unc/lib/dl/util/DepositStatusFactoryTest.java (hunk 12 above)
Comment (harringj) [inline metadata/src/test/java/edu/unc/lib/dl/util/DepositStatusFactoryTest.java]: Added

File: metadata/src/test/java/edu/unc/lib/dl/util/DepositStatusFactoryTest.java (hunk 11 above)
Comment (harringj) [inline metadata/src/test/java/edu/unc/lib/dl/util/DepositStatusFactoryTest.java]: Changes have been made

**Pull request merged**